### DISCUSSION TITLE
Is the Official Mastodon Self-Install UP TO DATE?

### DISCUSSION THREAD

**Discussion by TruthTriumphs** [mastodon/mastodon#24941] [Q&A] [answered]
**I tried the manual install from the Mastodon docs but ran into unexpected responses in my terminal as I labored through the prescribed process for a Linode Debian 11 container.**
https://docs.joinmastodon.org/admin/install/
This after discovering that the Linode Marketplace Mastodon install is from February & is missing key pieces, while its own docs are missing steps.

**Is there an up-to-date page of steps in all the package installs for a virgin install on a virtual host Debian 11, specifically for Linode?**

Also, 
I have tried to find explanations for a Docker vs Non-Docker install but haven't found it in sufficient detail. I have chosen, in two different trials, Non-Docker for one, Docker for the other.  I came to the end of the process and it promised me I could open the rest of the installation in my browser at my domain.  But I'm getting dropped connections.

Where can I go to get the most up-to-date sequence of installs as of now, May 12?
Thank you for responses!

**Comment by TruthTriumphs** [comment]
Secondary Question:
I see instructions on a Mastodon server installation requiring an SMTP server.
But I don't see more in depth about this.

For instance, does the Email Server need to be Set up in the server BEFORE Mastodon can be completely installed and running for Log In?

Or can I install the Email server service or software after getting Mastodon up and log in-able?

**Comment by justdaj** [comment, marked as answer]
I'd say it's up to date with regard to the current version.  There's certainly a few areas need expanded and are slightly wrong in my experience.

I had the email service running elsewhere so it was already setup.  During Mastodon install it will ask for details of the email server, but I think you can type anything then skip.  The email server is used to communicate with admin {you} and new users as they setup.
